image/serde/src/lib.rs

```diff
@@ -28,7 +29,40 @@ impl<W: Write> ImageBundleWriter<W> {
 
     /// Write Image Bundle
     pub fn write(&mut self, image: &ImageBundle) -> anyhow::Result<()> {
-        self.writer.write_all(image.manifest.as_bytes())?;
+        // Add fields till the ECC vendor key descriptor.
+        let offset = offset_of!(ImageManifest, preamble);
+        let ptr = &image.manifest as *const _ as *const u8;
+        let slice = unsafe { std::slice::from_raw_parts(ptr, offset) };
+        self.writer.write_all(slice)?;
+
+        // Add the ECC vendor key descriptor.
+        let vendor_pub_key_info = &image.manifest.preamble.vendor_pub_key_info;
+        self.writer
+            .write_all(vendor_pub_key_info.ecc_key_descriptor.as_bytes())?;
+        // Add the ECC vendor public key hashes.
+        self.writer.write_all(
+            (&vendor_pub_key_info.ecc_pub_key_hashes)
+                [..vendor_pub_key_info.ecc_key_descriptor.key_hash_count as usize]
+                .as_bytes(),
+        )?;
+
+        // Add the LMS vendor key descriptor.
+        self.writer
+            .write_all(vendor_pub_key_info.lms_key_descriptor.as_bytes())?;
+        // Add the LMS vendor public key hashes.
+        self.writer.write_all(
+            (&vendor_pub_key_info.lms_pub_key_hashes)
+                [..vendor_pub_key_info.lms_key_descriptor.key_hash_count as usize]
+                .as_bytes(),
+        )?;
+
+        // Add the remaining fields of the preamble.
+        let start = offset_of!(ImagePreamble, vendor_ecc_pub_key_idx);
```

Comment:
I think I'd still prefer to write each of the fields out rather than creating raw pointers.

Reply:
That's the cleanest way but my worry is someone adding a new field to the FW image and forgetting to update this code. That said, I updated the code to add individual fields.